### DISCUSSION TITLE
How to use the Button?

### DISCUSSION THREAD

**Discussion by esmorun** [AvaloniaUI/Avalonia#5486] [Q&A] [answered]
I'm new to Avalonia and have been reading the [documentation](https://avaloniaui.net/docs/controls/button) for the Button control. But I can't get it to work.

I've tried to use the Command paramenter to bind it to a simple method or a ReactiveCommand in my view model, but this results in the button being disabled and unclickable:
`<Button Name="buttonClear" Command="{Binding ClearItems}">Clear</Button>`

The button is stil disabled if I set IsEnabled to True in my xaml:
`<Button Name="buttonClear" Command="{Binding ClearItems}" IsEnabled="True">Clear</Button>`

If I try to use the Click event, Visual Studio will throw an error that is beyond my capability to understand.
` <Button Name="buttonClear" Width="98" Margin="2" Click="buttonClear_Click">Clear</Button>`

> Unable to find suitable setter or adder for property Click of type Avalonia.Controls:Avalonia.Controls.Button for argument System.Private.CoreLib:System.String, available setter parameter lists are:
System.EventHandler`1[[Avalonia.Interactivity.RoutedEventArgs, Avalonia.Interactivity, 

What am I doing wrong? Please help.

**Comment by FoggyFinder** [comment, marked as answer]
Can you show how `ClearItems` / `buttonClear_Click` look like?

**Reply by esmorun** [reply]
Thank you for the quick response. Here you go:
```
void ClearItems()
{
	MainWindow.ShowInformationBox("You clicked the Clear button");
}

private void buttonClear_Click(object sender, RoutedEventArgs e)
{
	MainWindow.ShowInformationBox("You clicked the Clear button");
}
```

I've also tried creating a ReactiveCommand like this:
```
public ReactiveCommand<Unit, Unit> buttonClear_Click { get; }

public ListBoxViewModel()
{
	buttonClear_Click = ReactiveCommand.Create(ClearItems);
}

private void ClearItems()
{
	MainWindow.ShowInformationBox("You clicked the Clear button");
}
```
These where all from a .cs file in my ViewModels-directory, my project is based on the MVVM template in Visual Studio if that matters.

**Reply by FoggyFinder** [reply]
use `public` access modifier

```csharp
public void ClearItems()
{
    // ...
}
```

```xml
Command="{Binding ClearItems}"
```

**Reply by esmorun** [reply]
I tried that now, but unfortunately the button is stil disabled both in the xaml preview window and when I launch the app :(

**Reply by FoggyFinder** [reply]
that's odd. Can you share your project via github or archive? I'll take a look.

**Reply by esmorun** [reply]
Thanks, its just a simple test project. I hope it's nothing more than a rookie mistake:
[https://www.dropbox.com/s/z34bdmrxjmwqv91/AvaloniaApplication1.zip?dl=0](https://www.dropbox.com/s/z34bdmrxjmwqv91/AvaloniaApplication1.zip?dl=0)

**Reply by FoggyFinder** [reply]
yeah - you just forgot to set `DataContext` for `ListBoxView`. See [The DataContext](https://avaloniaui.net/docs/binding/datacontext) page for details.

Quick fix would be

```csharp
public ListBoxView()
{
    InitializeComponent();
    DataContext = new ListBoxViewModel();
}
```

you can replace it with proper navigation later

**Reply by esmorun** [reply]
It works! Thank you so much! 👍
